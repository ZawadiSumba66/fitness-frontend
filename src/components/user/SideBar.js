import { slide as Menu } from 'react-burger-menu';
import PropTypes from 'prop-types';
import { Link } from '@reach/router';

const SideBar = ({ username, image }) => {
  const logOut = () => localStorage.clear();

  if (!username) {
    return <h1>Loading...</h1>;
  }
  return (
    <Menu className="d-flex flex-column bg-white justify-content-between">
      <div className="pb-5 pt-2 pl-2">
        <i className="fas fa-heartbeat" />
        <span className="font-weight-bold">KEEP IT FIT</span>
      </div>
      <div className="d-flex flex-column mb-5 text-center">
        <div>
          <img
            src={image}
            alt=""
            className="d-block mx-auto card-rounded-image"
          />
          <h5 className="my-2 text-center user-name menu-item text-orange">
            @
            {username}
          </h5>
        </div>
        <div className="d-flex flex-column">
          <Link
            to="/dashboard"
            className="text-dark font-weight-bold text-decoration-none text-hover my-3"
          >
            Dashboard
          </Link>
          <Link to="/tips" className="text-dark font-weight-bold text-decoration-none text-hover">
            View Tips
          </Link>
          <Link
            to="/create-tip"
            className="text-dark font-weight-bold text-decoration-none text-hover my-3"
          >
            Create Tip
          </Link>
        </div>
      </div>
      <div>
        <hr className="mb-2" />
        <Link
          to="/login"
          onClick={logOut}
          className="button-orange text-white py-1 font-weight-bold w-50 mx-auto text-decoration-none d-flex justify-content-center"
        >
          Logout
        </Link>
      </div>
    </Menu>
  );
};

SideBar.propTypes = {
  username: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default SideBar;
