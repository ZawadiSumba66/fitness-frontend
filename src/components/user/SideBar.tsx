import { slide as Menu } from 'react-burger-menu';
import { Link } from '@reach/router';
import React from 'react';

type UserSidebar = {
  username: string,
};

const SideBar: React.FunctionComponent<any> = ({ username }: UserSidebar) => {
  const logOut = () => localStorage.clear();

  return (
    <Menu className="d-flex flex-column bg-white justify-content-between">
      <div className="pb-5 pt-2 pl-2">
        <i className="fas fa-heartbeat" />
        <span className="font-weight-bold">KEEP IT FIT</span>
      </div>
      <div className="d-flex flex-column mb-5 text-center">
        <div>
          <h5 className="my-2 text-center user-name menu-item text-orange">
            @
            {username}
          </h5>
        </div>
        <div className="d-flex flex-column">
          <Link
            to="/dashboard"
            className="text-dark font-weight-bold text-decoration-none text-hover my-3"
          >
            Dashboard
          </Link>
          <Link to="/tips" className="text-dark font-weight-bold text-decoration-none text-hover">
            View Tips
          </Link>
          <Link
            to="/create-tip"
            className="text-dark font-weight-bold text-decoration-none text-hover my-3"
          >
            Create Tip
          </Link>
        </div>
      </div>
      <div>
        <hr className="mb-2" />
        <Link
          to="/login"
          onClick={logOut}
          className="button-orange text-white py-1 font-weight-bold w-50 mx-auto text-decoration-none d-flex justify-content-center"
        >
          Logout
        </Link>
      </div>
    </Menu>
  );
};

export default SideBar;
