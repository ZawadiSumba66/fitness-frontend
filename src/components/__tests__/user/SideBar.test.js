import renderer from 'react-test-renderer';
import SideBar from '../../user/SideBar';

it('matches the Sidebar componenent snapshot', () => {
  const tree = renderer.create(<SideBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
