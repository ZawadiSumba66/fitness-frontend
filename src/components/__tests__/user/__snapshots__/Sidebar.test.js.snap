// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the Sidebar componenent snapshot 1`] = `
<div>
  <div
    className="bm-overlay"
    onClick={[Function]}
    style={
      Object {
        "MozTransform": "translate3d(100%, 0, 0)",
        "MsTransform": "translate3d(100%, 0, 0)",
        "OTransform": "translate3d(100%, 0, 0)",
        "WebkitTransform": "translate3d(100%, 0, 0)",
        "background": "rgba(0, 0, 0, 0.3)",
        "height": "100%",
        "opacity": 0,
        "position": "fixed",
        "transform": "translate3d(100%, 0, 0)",
        "transition": "opacity 0.3s, transform 0s 0.3s",
        "width": "100%",
        "zIndex": 1000,
      }
    }
  />
  <div
    style={Object {}}
  >
    <div
      className="bm-burger-button"
      style={
        Object {
          "zIndex": 1000,
        }
      }
    >
      <button
        id="react-burger-menu-btn"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        style={
          Object {
            "background": "transparent",
            "border": "none",
            "cursor": "pointer",
            "fontSize": 0,
            "height": "100%",
            "left": 0,
            "margin": 0,
            "padding": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
            "zIndex": 1,
          }
        }
      >
        Open Menu
      </button>
      <span>
        <span
          className="bm-burger-bars"
          style={
            Object {
              "height": "20%",
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": "0%",
            }
          }
        />
        <span
          className="bm-burger-bars"
          style={
            Object {
              "height": "20%",
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": "40%",
            }
          }
        />
        <span
          className="bm-burger-bars"
          style={
            Object {
              "height": "20%",
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": "80%",
            }
          }
        />
      </span>
    </div>
  </div>
  <div
    aria-hidden={true}
    className="bm-menu-wrap d-flex flex-column bg-white justify-content-between"
    id=""
    style={
      Object {
        "MozTransform": "translate3d(-100%, 0, 0)",
        "MsTransform": "translate3d(-100%, 0, 0)",
        "OTransform": "translate3d(-100%, 0, 0)",
        "WebkitTransform": "translate3d(-100%, 0, 0)",
        "height": "100%",
        "position": "fixed",
        "right": "inherit",
        "transform": "translate3d(-100%, 0, 0)",
        "transition": "all 0.5s",
        "width": "300px",
        "zIndex": 1100,
      }
    }
  >
    <div
      className="bm-menu"
      style={
        Object {
          "boxSizing": "border-box",
          "height": "100%",
          "overflow": "auto",
        }
      }
    >
      <nav
        className="bm-item-list"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div
          className="bm-item pb-5 pt-2 pl-2"
          style={
            Object {
              "display": "block",
            }
          }
          tabIndex={-1}
        >
          <i
            className="fas fa-heartbeat"
          />
          <span
            className="font-weight-bold"
          >
            KEEP IT FIT
          </span>
        </div>
        <div
          className="bm-item d-flex flex-column mb-5 text-center"
          style={
            Object {
              "display": "block",
            }
          }
          tabIndex={-1}
        >
          <div>
            <img
              className="d-block mx-auto max-h-52 rounded-full"
            />
            <h5
              className="my-2 text-center user-name menu-item text-orange"
            >
              @
            </h5>
          </div>
          <div
            className="d-flex flex-column"
          >
            <a
              className="text-dark font-weight-bold text-decoration-none text-hover my-3"
              href="/dashboard"
              onClick={[Function]}
            >
              Dashboard
            </a>
            <a
              className="text-dark font-weight-bold text-decoration-none text-hover"
              href="/tips"
              onClick={[Function]}
            >
              View Exercises
            </a>
            <a
              className="text-dark font-weight-bold text-decoration-none text-hover my-3"
              href="/create-tip"
              onClick={[Function]}
            >
              Create an Exercise Tip
            </a>
          </div>
        </div>
        <div
          className="bm-item"
          style={
            Object {
              "display": "block",
            }
          }
          tabIndex={-1}
        >
          <hr
            className="mb-2"
          />
          <a
            className="button-orange text-white py-1 font-weight-bold w-50 mx-auto text-decoration-none d-flex justify-content-center"
            href="/login"
            onClick={[Function]}
          >
            Logout
          </a>
        </div>
      </nav>
    </div>
    <div
      style={Object {}}
    >
      <div
        className="bm-cross-button"
        style={
          Object {
            "height": 24,
            "position": "absolute",
            "right": 8,
            "top": 8,
            "width": 24,
          }
        }
      >
        <button
          id="react-burger-cross-btn"
          onClick={[Function]}
          style={
            Object {
              "background": "transparent",
              "border": "none",
              "cursor": "pointer",
              "fontSize": 0,
              "height": "100%",
              "left": 0,
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
              "zIndex": 1,
            }
          }
          tabIndex={-1}
        >
          Close Menu
        </button>
        <span
          style={
            Object {
              "position": "absolute",
              "right": "14px",
              "top": "6px",
            }
          }
        >
          <span
            className="bm-cross"
            style={
              Object {
                "height": 14,
                "position": "absolute",
                "transform": "rotate(45deg)",
                "width": 3,
              }
            }
          />
          <span
            className="bm-cross"
            style={
              Object {
                "height": 14,
                "position": "absolute",
                "transform": "rotate(-45deg)",
                "width": 3,
              }
            }
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
